include/mbedtls/error.h

```diff
@@ -114,6 +119,54 @@ extern "C" {
 #define MBEDTLS_ERR_ERROR_GENERIC_ERROR       -0x0001  /**< Generic error */
 #define MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED -0x006E  /**< This is a bug in the library */
 
+/**
+ * \brief Combines a high-level and low-level error code together.
+ *
+ *        Wrapper function for mbedtls_err_add_ext(). See that function for
+ *        more details.
+ */
+#define MBEDTLS_ERROR_ADD( high, low ) \
+        mbedtls_error_add_ext( high, low, __FILE__, __LINE__ )
+
+#if defined(MBEDTLS_TEST_HOOKS)
+/**
+ * \brief Testing hook called before adding/combining two error codes together.
+ *        Only used when invasive testing is enabled via MBEDTLS_TEST_HOOKS.
+ */
+extern void (*mbedtls_test_hook_error_add)( int, int, const char *, int );
```

Comment:
I think it was done because the testing framework has visibility/access to the library but not the other way around. This way the testing module sets it's own checking function to be called when an addition occurs and `error.h` does not need to include test specific code.

It also means that it will still run (without checks) for users who do not run our testing framework unless they wanted to use their own checking function.

Whether this is actually useful and something we want is beyond me so if you still disagree @gilles-peskine-arm may have a better idea of why it's important.